tests/array/good/fold.catala_en

```diff
@@ -1,35 +1,82 @@
 ## Article
 
 ```catala
-declaration structure S:
-  data id content integer
-  data income content money
-
 declaration scope A:
-   context output x content list of S
+  output x content list of money
+  output y content list of decimal
 
 scope A:
-   definition x equals [
-      S { -- id: 0 -- income: $0 };
-      S { -- id: 1 -- income: $4 + $5 };
-      S { -- id: 2 -- income: $8 * 0.65 }
-   ]
+   definition x equals [$12; $13.50; $11; $100.1; $50.20]
+   definition y equals [1.0; 2.0; 3.0; 4.0; 5.9]
 
 declaration scope B:
-   a scope A
-   output argmax content S
-   output argmin content S
+  a scope A
+  output count content integer
+  output max content money
+  output min content money
+  output total content money
+  output reverse content list of money
+  output ratios1 content (list of money, money)
+  output ratios2 content money
+  output ratios3 content (list of money, money)
+  output sorted content list of money
+
+declaration insert content list of money
+  depends on element content money,
+             lst content list of money
+  equals
+  let (lower, higher) equals
+    for (lower, higher) initially ([], []) and then
+      (if x <= element then (lower ++ [x], higher) else (lower, higher ++ [x]))
+      for x among lst
+  in
+  lower ++ [element] ++ higher
```

Comment:
HAHAHA that's it we can program insertion sort in Catala, we're doomed !

Reply:
My thoughts exactly when I was writing it ;)